### PR TITLE
que1.6

### PR DESCRIPTION
**Pull request by ashok-choudhary-0**

### CODE REVIEW

File: controllers/userController.js [hunk 1]
```diff
@@ -68,5 +68,14 @@ const deleteUserData = async (req, res) => {
     res.status(500).send({ message: err })
   }
 }
+const getAllUsersData = async (req, res) => {
+  const { page } = req.params
+  try {
+    const userData = await userModel.findAll({ limit: 10, offset: (page - 1) * 10 })
```
Comment (amrit56789) [inline controllers/userController.js]: Why you use here limit 10 suppose If I want only 5 user of 1st page then what to do?

File: controllers/userController.js [hunk 2]
```diff
@@ -68,5 +68,14 @@ const deleteUserData = async (req, res) => {
     res.status(500).send({ message: err })
   }
 }
+const getAllUsersData = async (req, res) => {
```
Comment (amrit56789) [inline controllers/userController.js]: we are find limit userdata so choose variable name carefully.

File: controllers/userController.js [hunk 3]
```diff
@@ -68,5 +68,19 @@ const deleteUserData = async (req, res) => {
     res.status(500).send({ message: err })
   }
 }
+const limitUsersData = async (req, res) => {
+  const page = req.params.page
+  const limitValue = Number(req.query.limit || 10)
```
Comment (amrit56789) [inline controllers/userController.js]: Limit also go in params

File: controllers/userController.js [hunk 4]
```diff
@@ -68,5 +68,19 @@ const deleteUserData = async (req, res) => {
     res.status(500).send({ message: err })
   }
 }
+const limitUsersData = async (req, res) => {
+  const page = req.params.page
```
Comment (meanstack9716) [inline controllers/userController.js]: you should also parse/number the page param as well as @ashok-choudhary-0

File: controllers/userController.js [hunk 5]
```diff
@@ -68,5 +68,19 @@ const deleteUserData = async (req, res) => {
...
+  const limitValue = Number(req.query.limit) || 10
+  try {
+    const userData = await userModel.findAndCountAll({ limit: limitValue, offset: (page - 1) * limitValue })
+    if (userData.rows.length != 0) {
+      res.status(200).send(userData)
+    } else {
+      res.status(500).send({ message: "oops data not found" })
```
Comment (meanstack9716) [inline controllers/userController.js]: do not send 500 status without error, 500 is for there is some issue in code, that server can't handle.
So this response will also be 200 with count 0 and user data blank array @ashok-choudhary-0

**Review by meanstack9716 — CHANGES_REQUESTED** [review]

**Pull request merged**